src/main/java/org/terasology/anotherWorld/decorator/BeachDecorator.java

```diff
@@ -56,12 +57,12 @@ public void initialize() {
     public void generateChunk(CoreChunk chunk, Region chunkRegion) {
         SurfacesFacet surfacesFacet = chunkRegion.getFacet(SurfacesFacet.class);
         TerrainVariationFacet terrainVariationFacet = chunkRegion.getFacet(TerrainVariationFacet.class);
-        for (Vector3i position : chunk.getRegion()) {
-            for (int groundLevel : surfacesFacet.getWorldColumn(position.x, position.z)) {
+        for (Vector3ic position : chunk.getRegion()) {
+            for (int groundLevel : surfacesFacet.getWorldColumn(position.x(), position.z())) {
                 if (groundLevel <= toLevel && groundLevel >= fromLevel) {
                     for (int y = fromLevel; y <= toLevel; y++) {
-                        if (blockFilter.apply(chunk.getBlock(ChunkMath.calcRelativeBlockPos(position)))) {
-                            chunk.setBlock(ChunkMath.calcRelativeBlockPos(position), beachBlockProvider.provide(terrainVariationFacet.get(position.x, position.y, position.z)));
+                        if (blockFilter.apply(chunk.getBlock(ChunkMath.calcRelativeBlockPos(position, new Vector3i())))) {
+                            chunk.setBlock(ChunkMath.calcRelativeBlockPos(position, new Vector3i()), beachBlockProvider.provide(terrainVariationFacet.get(position.x(), position.y(), position.z())));
```

Comment:
One could try to safe one vector allocation here - not sure whether that's worth it... I think it would get less readable like that...
```suggestion
						Vector3i tmp = ChunkMath.calcRelativeBlockPos(position, new Vector3i());
                        if (blockFilter.apply(chunk.getBlock(tmp))) {
                            chunk.setBlock(ChunkMath.calcRelativeBlockPos(position, tmp), beachBlockProvider.provide(terrainVariationFacet.get(position.x(), position.y(), position.z())));
```